frameworks/keyed/mogwai/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+    label: Model<String>,
+}
+
+static ID_COUNTER: AtomicUsize = AtomicUsize::new(1);
+
+// ------ ------
+//    Message
+// ------ ------
+
+#[derive(Clone)]
+enum Msg {
+    Create(Count),
+    Append(Count),
+    Update(Step),
+    Clear,
+    Swap,
+    Select(Id),
+    Remove(Id),
+}
+
+// ------ ------
+//    Update
+// ------ ------
+
+async fn update(msg: Msg, mdl: &Mdl) {
+    match msg {
+        Msg::Create(cnt) => {
+            mdl.selected.visit_mut(|s| *s = None).await;
+            let new_rows = build_data(cnt);
+            for p in new_rows.into_iter().map(ListPatch::push) {
+                mdl.rows.patch(p).await.expect("could not patch");
+            }
+        }
+        Msg::Append(cnt) => {
+            let new_rows = build_data(cnt);
+            for p in new_rows.into_iter().map(ListPatch::push) {
+                mdl.rows.patch(p).await.expect("could not patch");
+            }
+        }
+        Msg::Update(step_size) => {
+            let rows = mdl.rows.read().await;
+            for row in rows.iter().step_by(step_size) {
+                row.label.visit_mut(|l| l.push_str(" !!!")).await;
+            }
+        }
+        Msg::Clear => {
+            mdl.selected.visit_mut(|s| *s = None).await;
+            let patch = ListPatch::drain();
+            mdl.rows.patch(patch).await.expect("could not patch");
+        }
+        Msg::Swap => {
+            // TODO
+        }
+        Msg::Select(id) => {
+            mdl.selected.visit_mut(|s| *s = Some(id)).await;
+        }
+        Msg::Remove(remove_id) => {
+            let rows = mdl.rows.read().await;
+            let index = rows
+                .iter()
+                .enumerate()
+                .find_map(|(i, row)| if row.id == remove_id { Some(i) } else { None })
+                .unwrap();
+            drop(rows);
+            let patch = ListPatch::remove(index);
+            mdl.rows.patch(patch).await.expect("could not patch");
+        }
+    }
+}
+
+fn build_data(count: usize) -> Vec<Row> {
+    let mut thread_rng = thread_rng();
+
+    let mut data = Vec::new();
+    data.reserve_exact(count);
+
+    for _ in 0..count {
+        let adjective = ADJECTIVES.choose(&mut thread_rng).unwrap();
+        let colour = COLOURS.choose(&mut thread_rng).unwrap();
+        let noun = NOUNS.choose(&mut thread_rng).unwrap();
+        let capacity = adjective.len() + colour.len() + noun.len() + 2;
+        let mut label = String::with_capacity(capacity);
+        label.push_str(adjective);
+        label.push(' ');
+        label.push_str(colour);
+        label.push(' ');
+        label.push_str(noun);
+
+        data.push(Row {
+            id: ID_COUNTER.load(Ordering::Relaxed),
+            label: Model::new(label),
+        });
+
+        ID_COUNTER.store(ID_COUNTER.load(Ordering::Relaxed) + 1, Ordering::Relaxed);
+    }
+
+    data
+}
+
+// ------ ------
+//     View
+// ------ ------
+
+fn view(mdl: &Mdl) -> (ViewBuilder, impl Stream<Item = Msg>) {
```

Comment:
Instead of returning `(ViewBuilder, impl Stream)` it might be more succinct to return `ViewBuilder` and add the task to the end of `rsx!`.